configuration/go/http/order-processor/app.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+var DAPR_HOST, DAPR_HTTP_PORT string
+var okHost, okPort bool
+var appPort = "6001"
+var DAPR_CONFIGURATION_STORE = "configstore"
+var CONFIGURATION_ITEMS = []string{"orderId1", "orderId2"}
+
+func main() {
+	if DAPR_HOST, okHost = os.LookupEnv("DAPR_HOST"); !okHost {
+		DAPR_HOST = "http://localhost"
+	}
+	if DAPR_HTTP_PORT, okPort = os.LookupEnv("DAPR_HTTP_PORT"); !okPort {
+		DAPR_HTTP_PORT = "3500"
+	}
+	if value, ok := os.LookupEnv("APP_PORT"); ok {
+		appPort = value
+	}
+
+	// Get config items from the config store
+	for _, item := range CONFIGURATION_ITEMS {
+		getResponse, err := http.Get(DAPR_HOST + ":" + DAPR_HTTP_PORT + "/v1.0-alpha1/configuration/" + DAPR_CONFIGURATION_STORE + "?key=" + item)
+		if err != nil {
+			fmt.Print("Could not get config item, err:" + err.Error())
+			os.Exit(1)
+		}
+		result, _ := ioutil.ReadAll(getResponse.Body)
+		fmt.Println("Configuration for "+item+":", string(result))
+	}
+
+	// Create POST endpoint to receive config updates
+	go startServerToListen()
+	// Subscribe for config updates
+	go subscribeToConfigUpdates()
```

Comment:
does this need to be a go routine?

Reply:
starting the http server blocks the main goroutine, hence running it on a separete goroutine